### PR TITLE
Change slasher DB structure to mirror beacon-chains 

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR changes the DB structure of the slasher to mirror the beacon chains DB, including a `kv`, `testing`, and `iface` package.

No functionality changes.

### CODE REVIEW

File: slasher/db/iface/interface.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import (
+	"io"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	slashpb "github.com/prysmaticlabs/prysm/proto/slashing"
+	"github.com/prysmaticlabs/prysm/slasher/db/types"
+)
+
+// SlasherDB defines the necessary methods for a Prysm slasher DB.
+type SlasherDB interface {
```
Comment (terencechain) [inline slasher/db/iface/interface.go]: Would be good to separate this to read only and read + write

File: slasher/db/types/types.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package types
```
Comment (terencechain) [inline slasher/db/types/types.go]: Why is types under `/slasher/db`, do you think it should be under `/slasher`?

File: slasher/db/iface/interface.go (hunk 1 above)
Comment (nisdas) [reply slasher/db/iface/interface.go]: +1 , define different interfaces for different access patterns

File: slasher/db/types/types.go (hunk 2 above)
Comment (0xKiwi) [reply slasher/db/types/types.go]: Should I move all types in there then? Types like Service, Config, Server?

File: slasher/db/kv/kv.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/dgraph-io/ristretto"
+	"github.com/pkg/errors"
+)
+
+var databaseFileName = "slasher.db"
+
+// Store defines an implementation of the Prysm Database interface
```
Comment (rauljordan) [inline slasher/db/kv/kv.go]:
```suggestion
// Store defines an implementation of the slasher Database interface
```

File: slasher/db/testing/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,23 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    testonly = True,
+    srcs = ["setup_db.go"],
+    importpath = "github.com/prysmaticlabs/prysm/slasher/db/testing",
+    visibility = ["//visibility:public"],
```
Comment (rauljordan) [inline slasher/db/testing/BUILD.bazel]:
```suggestion
    visibility = ["//slasher:__subpackages__"],
```

File: slasher/db/types/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["types.go"],
+    importpath = "github.com/prysmaticlabs/prysm/slasher/db/types",
+    visibility = ["//visibility:public"],
```
Comment (rauljordan) [inline slasher/db/types/BUILD.bazel]:
```suggestion
    visibility = ["//slasher:__subpackages__"],
```

File: slasher/db/types/types.go (hunk 2 above)
Comment (rauljordan) [reply slasher/db/types/types.go]: These types are highly specific only to the db, I think we should just leave them as a subpackage under `slasher/db/types` @terencechain

File: slasher/db/types/types.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+type SlashingType uint8
+
+const (
+	// Proposal enum value.
+	Proposal = iota
+	// Attestation enum value.
+	Attestation
+)
+
+// String returns the string representation of the status SlashingType .
```
Comment (rauljordan) [inline slasher/db/types/types.go]:
```suggestion
// String returns the string representation of the status SlashingType.
```

**Pull request merged**